src/types/ErrorRate.ts

```diff
@@ -0,0 +1,335 @@
+          }
+          tol.requests[protocol].requestRate += Number(value);
+          if (checkExpr(tol!.tolerance!.code, code)) {
+            tol.requests[protocol].errorRate += Number(value);
+          }
+        }
+      }
+    } else {
+      for (let [code, value] of Object.entries(req)) {
+        if (!Object.keys(tol.requests).includes(protocol)) {
+          tol.requests[protocol] = emptyRate();
+        }
+        tol.requests[protocol].requestRate += Number(value);
+        if (Object.keys(regexProtocol).includes(protocol) && regexProtocol[protocol].test(code)) {
+          tol.requests[protocol].errorRate += Number(value);
+        }
+      }
+    }
+    if (Object.keys(tol.requests).includes(protocol)) {
+      if (tol.requests[protocol].requestRate === 0) {
+        tol.requests[protocol].errorRatio = -1;
+      } else {
+        tol.requests[protocol].errorRatio = tol.requests[protocol].errorRate / tol.requests[protocol].requestRate;
+      }
+    }
+  }
+};
+
+const aggregate = (request: RequestType, tolerances?: ToleranceConfig[]): RequestTolerance[] => {
+  let result: RequestTolerance[] = [];
+  if (request) {
+    if (tolerances) {
+      for (let tol of Object.values(tolerances)) {
+        let newReqTol: RequestTolerance = { tolerance: tol, requests: {} };
+        generateRateForTolerance(newReqTol, request);
+        result.push(newReqTol);
+      }
+    } else {
+      const emptyRequest: { [key: string]: Rate } = {};
+      for (let protocol of Object.keys(request)) {
+        emptyRequest[protocol] = emptyRate();
+      }
+      let newReqTol: RequestTolerance = { tolerance: undefined, requests: emptyRequest };
+      generateRateForTolerance(newReqTol, request);
+      result.push(newReqTol);
+    }
+  }
+  return result;
+};
+
+const emptyRate = (): Rate => {
+  return { requestRate: 0, errorRate: 0, errorRatio: 0 };
+};
+
+// Graph Edge
+
+export const graphEdgeColor = (edge: DecoratedGraphEdgeData): ThresholdStatus => {
+  if (edge && edge.responses) {
+    const cy = store.getState().graph.summaryData!.summaryTarget as Cy.Core;
+    const source = decoratedNodeData(cy.$id(edge.source));
+    const target = decoratedNodeData(cy.$id(edge.target));
+
+    // We need to check the configuration for item A outbound requests and configuration of B for inbound requests
+    const configSource = getConfig(source.namespace, source[source.nodeType], source.nodeType);
+    const configTarget = getConfig(target.namespace, target[source.nodeType], target.nodeType);
+
+    // If there is not tolerances with this configuration we'll use defaults
+    const tolerancesSource = configSource?.tolerance.filter(tol => checkExpr(tol.direction, 'outbound'));
+    const tolerancesTarget = configTarget?.tolerance.filter(tol => checkExpr(tol.direction, 'inbound'));
+
+    // Calculate aggregate
+    const outboundEdge = aggregateGraph(
+      transformEdgeResponses(edge.responses, edge.hasTraffic ? edge.protocol : '*'),
```

Comment:
With the suggested addition above of a hasTraffic check in the initial guard, you can assume edge.protocol is set.